tox.ini

```diff
@@ -1,19 +1,27 @@
 [tox]
 minversion=2.3.1
 envlist =
-    py{27,35}-ansible22-{pylint,unit,flake8,yamllint,generate_validation}
+    py{27,35}-ansible22-{lint,unit, functional}
 skipsdist=True
 skip_missing_interpreters=True
 
 [testenv]
 deps =
     -rtest-requirements.txt
-    py35-flake8: flake8-bugbear
-    ansible22: ansible~=2.2
+    py35-lint: flake8-bugbear
+    ansible22: ansible==2.2.0.0
 
 commands =
-    flake8: flake8
-    pylint: python setup.py lint
-    yamllint: python setup.py yamllint
+    lint: flake8
+    lint: python setup.py lint
+    lint: python setup.py yamllint
+    lint: python setup.py generate_validation
```

Comment:
@rhcarvalho yes, generate_validation does indeed validate generated code is up to date.  I'm open to renaming.

I'm working on trying to do two things with the env reworking:
- improve travis parallelization
- improve general runtimes, I found that by adding wheel as a dependency then we are able to share pre-built packages from the pip cache.